SAP/spartacus
# feat: CXSPA-5713 Upgrade to Angular 17 - schematics for removing 'provideClientHydration' from AppModule

This PR contains schematics for removing `provideClientHydration` from AppModule `providers` in the fresh app.
This feature has to be removed, because Spartacus doesn't support it yet and it may cause unexpected issues.

QA steps:

- build and deploy Spartacus packages locally: 
`npx ts-node ./tools/schematics/testing.ts` in the SPA root folder
- create fresh ng17 app: 
`npx @angular/cli new my-app --standalone=false --ssr=false`
- go to project's directory and install SPA with SSR: 
`npx @angular/cli add @spartacus/schematics@latest --baseUrl="https://40.76.109.9:9002" --ssr --debug`
- verify whether `provideClientHydration` has been removed from `providers` in AppModule together with its import.
-build and run the app to verify if it works as expected: 
`npm run build` -> `npm run serve:ssr:my-app`

closes [CXSPA-5713](https://jira.tools.sap/browse/CXSPA-5713)
